duo-client/src/test/java/com/duosecurity/client/HttpCanonRequestTest.java

```diff
@@ -57,6 +57,17 @@ private void assertCanonRequest(String expected_query_string,
                                  + expected_prefix
                                  + expected_query_string),
                                 actual);
+
+            try {
+                h.canonRequest(date, 99);
+                Assert.fail("Expected exception was not thrown");
+            }
+            catch (IllegalArgumentException e){
+                Assert.assertEquals(new IllegalArgumentException("Invalid Request Signature").toString(), e.toString());
```

Comment:
Not sure if this is the right way to do java test. Found some example online and this just works.